src/requests/sessions.py

```
@@ -514,8 +514,19 @@ def request(
         stream=None,
         verify=None,
         cert=None,
-        json=None,
+        retry_on_failure=False
     ):
+        """Send a request. :param retry_on_failure: If True, retry once on certain connection-related errors before raising."""
+    try:         
+        # existing logic (unchanged) that prepares and sends
+        # the request and returns a response
+        return self._send(method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects,proxies, hooks, stream, verify, cert, json)
+    except (ConnectionError, Timeout):
```

Ruff (0.14.8)

Undefined name `Timeout`

(F821)